hydra-prelude/src/Hydra/Prelude.hs

```diff
@@ -235,3 +239,13 @@ infixl 4 ?>
   case m of
     Nothing -> Left e
     Just a -> Right a
+
+-- | Like 'withFile' from 'base', but without annotating errors originating from
+-- enclosed action.
+--
+-- XXX: This should be fixed upstream in 'base'.
+withFile :: FilePath -> IOMode -> (Handle -> IO a) -> IO a
+withFile fp mode action =
+  System.IO.withFile fp mode (try . action) >>= \case
+    Left (e :: IOException) -> throwIO e
```

Comment:
Aren't we losing the annotation of errors thrown by `withFile` proper? Perhaps they are useless anyway

Reply:
No. If there is a problem with a file not existing (for example), it catches and annotates that exception inside `System.IO.withFile`. The `>>=` will never be executed.

Reply:
Because `withFile` in `base` is not simply a reexport of `System.IO.withFile`?

Reply:
I am shocked